library/Rules/Uuid.php

```diff
@@ -34,18 +37,39 @@ final class Uuid extends AbstractRule
     /**
      * The UUID version to validate for.
      *
-     * @var int|null
      */
-    private $version;
+    private ?int $version;
+
+    /**
+     * Whether to use Ramsey/Uuid to validate the UUID.
+     *
+     */
+    private bool $useRamseyUuid = false;
+
+    /**
+     * Whether Ramsey/Uuid is loaded.
+     *
+     */
+    private static ?bool $ramseyUuidIsLoaded = null;
 
     /**
      * Initializes the rule with the desired version.
      *
      * @throws ComponentException when the version is not valid
      */
-    public function __construct(?int $version = null)
+    public function __construct(?int $version = null, ?bool $useRamseyUuid = null)
     {
+        if ($useRamseyUuid && !$this->ramseyUuidIsLoaded()) {
```

Comment:
That's really awesome, thank you so much! I would say though, we could aways use RamseyUuid for this rule, we just need to require it to be install. In that case, this branch should be based on `main`, to be released with the next major version. What do you say?

Reply:
I was trying not to introduce any BC breaks and allow RamseyUuid to be optional.

But if you are good with making RamseyUuid required, it would clean up the code loads.

I was targeting this at 2.4 so it could be used quicker, has I dont know how far version 3.0 is from release.

I will make the chages to make RamseyUuid a required and clean up the code 😁

Reply:
OK, just looking at the `main` version, I will create a new version for the `main` and leave the PR here if you want to merge this into `2.4`